### PR TITLE
Fix various assorted warnings

### PR DESCRIPTION
**Pull request by hpvb**
Fix various warnings that don't have enough instances to merit
individual commits. Also fixes a potential bug in audio_server.cpp.

### CODE REVIEW

File: core/io/resource_format_binary.cpp [hunk 1]
```diff
@@ -1814,11 +1814,11 @@ Error ResourceFormatSaverBinaryInstance::save(const String &p_path, const RES &p
 
 				if (skip_editor && F->get().name.begins_with("__editor"))
 					continue;
-				if (F->get().usage & PROPERTY_USAGE_STORAGE) {
+				if ((F->get().usage & PROPERTY_USAGE_STORAGE)) {
```
Comment (akien-mga) [inline core/io/resource_format_binary.cpp]: Why are the parentheses needed here?

Comment (hpvb) [reply core/io/resource_format_binary.cpp]: A 'suggestion' warning for GCC about comparing things in a binary context. I can remove the ones that are completely unambiguous to humans and disable the warning later though.

Comment (hpvb) [reply core/io/resource_format_binary.cpp]: I would prefer to keep the ones where its 'if (stuff & blah && something else)' as the single '&' and the double '&&' actually could be mistaken for each other.

Comment (akien-mga) [reply core/io/resource_format_binary.cpp]: Well of course it makes sense if `(stuff & blah) && blabla`, but here it's unambiguous so it seems weird that GCC would raise a warning. But if it does, I guess the extraneous parentheses don't hurt. I'm just pretty sure that they're useless as whoever adds a new conditional there might just do `if ((F->get().usage & PROPERTY_USAGE_STORAGE && new_cond))` :P

Comment (hpvb) [reply core/io/resource_format_binary.cpp]: Yeah, this one will require some upkeep probably. I don't mind doing this manually from time to time. I think we'll never be able to enable -Werror indiscriminately. I'm hoping we can get a subset of warnings always enabled with -Werror by default though. maybe unitialized, unused, set but unused, misleading indendtation. 

We'll discuss that when I make a PR to take a first stab at that.

**Pull request merged**